src/Shared/Microsoft.DotNet.Scaffolding.Shared/Project/ProjectModifierHelper.cs

```diff
@@ -420,14 +428,14 @@ internal static async Task<Document> ModifyDocumentText(Document fileDoc, IEnume
                 return null; // TODO generate README
             }
 
-            var sourceTextToAdd = SourceText.From(sourceFileString);
-            return fileDoc.WithText(sourceTextToAdd);
+            var updatedSourceText = SourceText.From(sourceFileString);
+            return fileDoc.WithText(updatedSourceText);
         }
 
         internal static async Task UpdateDocument(Document fileDoc, Document editedDocument, IConsoleLogger consoleLogger)
         {
             var classFileTxt = await editedDocument.GetTextAsync();
-            File.WriteAllText(fileDoc.Name, classFileTxt.ToString());
+            File.WriteAllText(fileDoc.FilePath, classFileTxt.ToString());
```

Comment:
double check the value for Document.Name vs Document.FilePath. I remember distinctly using Document.Name but blanking rn. 
Maybe add a comment showing the difference between their values.